frontend/src/components/pages/cabins/booking/Steps/CabinSelection.tsx

```diff
@@ -0,0 +1,205 @@
+import { useQuery } from "@apollo/client";
+import { KeyboardArrowRight } from "@mui/icons-material";
+import { Button, Checkbox, Divider, FormControlLabel, Paper, Stack, Tooltip, Typography } from "@mui/material";
+import dayjs from "dayjs";
+import React from "react";
+
+import { Calendar } from "@/components/Calendar";
+import { BookingSemesterDocument, CabinFragment } from "@/generated/graphql";
+import { useOverlappingBookings } from "@/hooks/cabins/useBookedDates";
+import { useResponsive } from "@/hooks/useResponsive";
+
+import { Stepper } from "./Stepper";
+
+type Props = {
+  allCabins: CabinFragment[];
+  chosenCabins: CabinFragment[];
+  onCabinsChange: (cabins: CabinFragment[]) => void;
```

Comment:
I looked through the codebase and searched for "onCabinsChange" but i couldn't find the definition of the function. Am i blind or is threre something interesting to learn?

Reply:
Short version: It's just the name of a prop we can pass to `CabinSelection`, so we can pass an arbitrary function - with an arbitrary name. In this case, the function that gets passed down is `setChosenCabins`, the state setting function. See [link](https://github.com/rubberdok/indok-web/blob/d2e40604c391a390161f5bf6c597723a4f070e7e/frontend/src/pages/cabins/book/index.tsx#L164).

**Motivation:** Previously, this prop was called `setChosenCabins`, but intentionally renamed to `onCabinsChange` in this PR to convey to the consumers of this component that the function that this function will get called when cabins are changed. That way, it's easier to understand how to use the component, rather than it just being called `setChosenCabins`, which doesn't convey if the function is called when cabins are deleted, updated, or changed in some other way.

Throughout the code base, you'll frequently encounter props whose name just reflect the name of the state setting function, i.e. `setXXX`. It would be better for them to have more descriptive names of what the functions should do. That is, instead of `setXXX`, call them stuff like "onPurchaseCompleted" or `onBookingRejection`. By doing that, you better convey to whoever uses your component in the future what to expect, and how to use the different props.